sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisRunContext.cs

```diff
@@ -35,5 +37,37 @@ public AnalysisRunContext(SyntaxTree syntaxTree, IEnumerable<DiagnosticDescripto
         public SyntaxTree SyntaxTree { get; }
 
         public IEnumerable<DiagnosticDescriptor> SupportedDiagnostics { get; }
+
+        // Real bug found in Roslyn - see https://github.com/dotnet/roslyn/pull/21258
+        public static bool TryRedirect(AssemblyName name, byte[] token, int major, int minor, int build, int revision)
+        {
+            var version = new Version(major, minor, revision, build);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Parameters to '.ctor' have the same names but not the same order as the method arguments. ([csharpsquid:S2234](https://sonarcloud.io/organizations/sonarsource/rules#rule_key=csharpsquid%3AS2234))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=sonaranalyzer-dotnet&pullRequest=1382&issues=AWM4IJJDmtQEh111wI_r&open=AWM4IJJDmtQEh111wI_r)

[ISSUE_KEY]: AWM4IJJDmtQEh111wI_r